Realm/RLMSyncUser.h

```diff
@@ -118,6 +131,39 @@ NS_SWIFT_UNAVAILABLE("Use the full version of this API.");
  */
 - (NSArray<RLMSyncSession *> *)allSessions;
 
+#pragma mark - Permissions
+
+/**
+ Asynchronously retrieve all permissions associated with the given user.
+
+ The results will be returned through the callback block, or an error if the operation fails.
+ */
+- (void)retrievePermissionsWithCallback:(RLMPermissionResultsBlock)callback;
+
+/**
+ Apply a given permission.
+
+ The operation will take place asynchronously, and the callback will be used to report whether
+ the permission change succeeded or failed. The user upon which this method is called must have
+ the right to grant the given permission, or else the oepration will fail.
+
+ @see `RLMSyncPermissionValue`
+ */
+- (void)applyPermission:(RLMSyncPermissionValue *)permission callback:(RLMPermissionStatusBlock)callback;
+
+/**
+ Revoke a given permission.
+
+ The operation will take place asynchronously, and the callback will be used to report whether
+ the permission change succeeded or failed. The user upon which this method is called must have
+ the right to grant the given permission, or else the oepration will fail.
+
+ @see `RLMSyncPermissionValue`
+ */
+- (void)revokePermission:(RLMSyncPermissionValue *)permission callback:(RLMPermissionStatusBlock)callback;
+
+#pragma mark - Permissions (old)
```

Comment:
We should come up with better wording for deprecated permissions APIs than simply "old".